cmd/thanos/receive.go

```diff
@@ -187,11 +187,11 @@ func runReceive(
 	if err != nil {
 		return err
 	}
-	rwTLSClientConfig, err := tls.NewClientConfig(logger, rwClientCert, rwClientKey, rwClientServerCA, rwClientServerName)
+	rwTLSClientConfig, err := tls.NewClientConfig(logger, rwClientCert, rwClientKey, rwClientServerCA, rwClientServerName, false)
```

Comment:
This shouldn’t blindly be disabled, doing proper TLS certificate provisioning should be encouraged by default.

Reply:
@brancz False there isn't blindly disabling it. It is disabling the skip verification. So it keeps the default behavior of validating certs. 

The reason I don't want to have to do real certs is that in order to generate certs automatically (say through ACM or K8s cert-mannager) it requires you use a public DNS zone. Meaning I would be putting entries like prometheus-0.company-domain.com that resolve to private IPs in a public zone. While that doesn't allow you to get to that server I have always been opposed to putting private resolution in public zones.

Reply:
@brancz I took your opinion and actually decided to go with a private CA and generate certs against that for the private domain, then just share the CA cert with Thanos.

In the end I still think this PR is useful for the case you mentioned - dev.

Reply:
oJ5vMr]ig-D72Yfc/JgV'Q&BA8)?TN5?

Reply:
That makes sense, but we still want this to be based on a flag that the user passes no?